### PR TITLE
Aggregation Command - Get User Data

### PR DESCRIPTION
**Pull request by mmhw**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
Related: [link to the issue](https://jira-dc.paloaltonetworks.com/browse/CIAC-11252)

## Description
Aggregation Command - Get User Data

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/CommonScripts/Scripts/GetUserData/GetUserData.py [hunk 1]
```diff
@@ -0,0 +1,570 @@
...
+                if user_email
+                else None,
+            },
+            is_generic=True,
+            output_key="Account",
+            output_function=lambda entry_context: create_account(
+                id=entry_context.get("id"),
+                type=entry_context.get("type"),
+            ),
+        )
```
Comment (bziser) [inline Packs/CommonScripts/Scripts/GetUserData/GetUserData.py:450]: I would move all this command creations to another function,
so the main would be easier and shorter to read

File: Packs/CommonScripts/Scripts/GetUserData/GetUserData.py [hunk 2]
```diff
@@ -0,0 +1,570 @@
...
+        aws_iam_get_user_task = Task(
+            "aws_iam_get_user_command", aws_iam_get_user_command
+        )
+        msgraph_user_get_task = Task(
+            "msgraph_user_get_command", msgraph_user_get_command
+        )
+        xdr_list_risky_users_task = Task(
+            "xdr_list_risky_users_command", xdr_list_risky_users_command
+        )
+        iam_get_user_task = Task("iam_get_user_command", iam_get_user_command)
```
Comment (bziser) [inline Packs/CommonScripts/Scripts/GetUserData/GetUserData.py:490]: Also the creation of the tasks, I would move to another function

File: Packs/CommonScripts/Scripts/GetUserData/GetUserData.py [hunk 3]
```diff
@@ -0,0 +1,582 @@
...
+        tasks = {"start_task": Task("start_task")}
+        tasks.update(
+            {
+                command_name.replace("_command", "_task"): Task(
+                    command_name.replace("_command", "_task"), command
+                )
+                for command_name, command in commands.items()
+            }
+        )
+        return tasks
```
Comment (bziser) [inline Packs/CommonScripts/Scripts/GetUserData/GetUserData.py:474]:
```suggestion
def setup_tasks(commands: Dict[str, Command]) -> Dict[str, Task]:
        tasks = {"start_task": Task("start_task")}
    for command_name, command in commands.items():
        task_name = command_name.replace("_command", "_task")
        tasks[task_name] = Task(task_name, command)
    return tasks
```

Comment (bziser) [reply Packs/CommonScripts/Scripts/GetUserData/GetUserData.py:474]: I think it is more readable

File: Packs/CommonScripts/Scripts/GetUserData/GetUserData.py [hunk 4]
```diff
@@ -0,0 +1,582 @@
...
+    def __init__(self):
+        self.tasks: dict[str, Task] = {}
+        self.connections: dict[str, list[tuple[str, Optional[Callable[[], bool]]]]] = {}
+        self.start_task_id: str = ""
+        self.outputs = []
+        self.results = []
+
+    def add_task(self, node: Task):
+        self.tasks[node.task_id] = node
+        self.connections[node.task_id] = []
```
Comment (bziser) [inline Packs/CommonScripts/Scripts/GetUserData/GetUserData.py:195]:
```suggestion
    def add_task(self, task: Task):
        self.tasks[task.task_id] = task
        self.connections[task.task_id] = []
```

File: Packs/CommonScripts/Scripts/GetUserData/GetUserData.py [hunk 5]
```diff
@@ -0,0 +1,582 @@
...
+
+    def add_tasks(self, tasks: list[Task]):
+        for node in tasks:
+            self.add_task(node)
```
Comment (bziser) [inline Packs/CommonScripts/Scripts/GetUserData/GetUserData.py:199]:
```suggestion
    def add_tasks(self, tasks: list[Task]):
        for task in tasks:
            self.add_task(task)
```

File: Packs/CommonScripts/Scripts/GetUserData/GetUserData.py [hunk 6]
```diff
@@ -0,0 +1,582 @@
...
+    def run(self):
+        current_task_ids = [self.start_task_id]
+        executed_task_ids = set()
+
+        while current_task_ids:
+            next_task_ids = []
+            for current_task_id in current_task_ids:
+                task = self.tasks.get(current_task_id)
+                if not task:
+                    continue
```
Comment (bziser) [inline Packs/CommonScripts/Scripts/GetUserData/GetUserData.py:218]:
```suggestion
                if not task:
                    demisto.debug(f"Task {current_task_id} not found.")
                    continue
```

File: Packs/CommonScripts/Scripts/GetUserData/GetUserData.py [hunk 7]
```diff
@@ -0,0 +1,582 @@
...
+            args_lst=self.args,
+        )
+        self.output_function = output_function
+        self.update_args_fun = update_args_fun
+
+    def _verify_args(self) -> bool:
+        if self.args:
+            return any(self.args.values())
+        else:
+            return True
```
Comment (bziser) [inline Packs/CommonScripts/Scripts/GetUserData/GetUserData.py:91]:
```suggestion
return any(self.args.values()) if self.args else True
```

**Comment by mmhw** [comment]
Replaced by https://github.com/demisto/content/pull/36222

**Pull request closed without merging**